cmd/talosctl/cmd/mgmt/cluster/create.go

```diff
@@ -436,6 +458,63 @@ func parseCPUShare() (int64, error) {
 	return nano.Num().Int64(), nil
 }
 
+func getDisks() ([]*provision.Disk, error) {
+	// should have at least a single primary disk
+	disks := []*provision.Disk{
+		{
+			Size: int64(clusterDiskSize) * 1024 * 1024,
+		},
+	}
+
+	for _, disk := range clusterDisks {
+		var (
+			partitions     = strings.Split(disk, ":")
+			diskPartitions = make([]*v1alpha1.DiskPartition, len(partitions)/2)
+			diskSize       int64
+		)
+
+		if len(partitions)%2 != 0 {
+			return nil, fmt.Errorf("failed to parse malformed partition definitions")
+		}
+
+		partitionIndex := 0
+
+		for j := 0; j < len(partitions); j += 2 {
+			partitionPath := partitions[j]
+
+			if !strings.HasPrefix(partitionPath, "/") {
+				return nil, fmt.Errorf("partition path should be absolute")
+			}
+
+			partitionSize, e := strconv.ParseInt(partitions[j+1], 10, -1)
+
+			if e != nil {
+				s, e := humanize.ParseBytes(partitions[j+1])
+
+				if e != nil {
+					return nil, fmt.Errorf("failed to parse partition size")
+				}
+
+				partitionSize = int64(s)
+			}
+
+			diskPartitions[partitionIndex] = &v1alpha1.DiskPartition{
+				DiskSize:       uint(partitionSize), // TODO: convert to int64
+				DiskMountPoint: partitionPath,
+			}
+			diskSize += partitionSize
+			partitionIndex++
+		}
+
+		disks = append(disks, &provision.Disk{
+			Size:       diskSize + 1024*1024,
```

Comment:
this is extra room for GPT (more than we need, but just in case, we might probably need some comment here to explain extra 1MiB)